src/Unleash/Scheduling/ClientMetricsBackgroundTask.cs

```diff
@@ -1,37 +1,35 @@
 ﻿using System;
-using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 using Unleash.Communication;
 using Unleash.Logging;
-using Unleash.Metrics;
+using Yggdrasil;
 
 namespace Unleash.Scheduling
 {
     internal class ClientMetricsBackgroundTask : IUnleashScheduledTask
     {
         private static readonly ILog Logger = LogProvider.GetLogger(typeof(ClientMetricsBackgroundTask));
-
+        private readonly YggdrasilEngine engine;
         private readonly IUnleashApiClient apiClient;
         private readonly UnleashSettings settings;
-        private readonly ThreadSafeMetricsBucket metricsBucket;
 
         public ClientMetricsBackgroundTask(
+            YggdrasilEngine engine,
             IUnleashApiClient apiClient, 
-            UnleashSettings settings,
-            ThreadSafeMetricsBucket metricsBucket)
+            UnleashSettings settings)
         {
+            this.engine = engine;
             this.apiClient = apiClient;
             this.settings = settings;
-            this.metricsBucket = metricsBucket;
         }
 
         public async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             if (settings.SendMetricsInterval == null)
                 return;
 
-            var result = await apiClient.SendMetrics(metricsBucket, cancellationToken).ConfigureAwait(false);
+            var result = await apiClient.SendMetrics(engine.GetMetrics(), cancellationToken).ConfigureAwait(false);
```

Comment:
I think this is okay for now, we'll probably have to do another pass later in a different PR to deal with the fact that this completely drops metrics if the HTTP call borks - the GetMetrics call to Ygg is destructive in that it clears the metrics once they've been read.

Might be a "Metrics Extensions v3" problem

Reply:
Any chance this is related to https://github.com/Unleash/unleash-client-dotnet/pull/224/files#r1660524625?

Reply:
Kinda sorta. It's adjacent certainly. But I'll put details on that comment rather. This can still wait, the other we should talk about